### PR TITLE
Enhancement: Use ondram/ci-detector to resolve build context

### LINKED ISSUE #1126: Use ondram/ci-detector to determine build context
We currently have our own implementation for determining the build context, as implemented in

- #1047
- #1051 
- #1062 

which are all related to #939.

We could use [`ondram/ci-detector`](https://github.com/OndraM/ci-detector/blob/master/composer.json) instead.

### Blocked by

- [x] ~https://github.com/OndraM/ci-detector/pull/55~
- [x] ~a corresponding release (see https://github.com/OndraM/ci-detector/releases/tag/3.3.0)~

### PR DESCRIPTION
**Pull request by localheinz**
This PR

* [x] requires `ondram/ci-detector`
* [x] implements a `CiDetectorResolver`
* [x] uses the `CiDetectorResolver`
* [x] removes unused resolvers
* [x] removes an unused parameter
* [x] removes an unnecessary interface
* [x] renames `CiDetectorResolver` to `BuildContextResolver`

Fixes #1126.
Replaces #939.

💁‍♂ In retrospect, https://github.com/OndraM/ci-detector/pull/55 was not necessary.

### CODE REVIEW

**Comment by localheinz** [comment]
@drupol 

Since this could replace #939, would you like to take a look as well?

**Comment by drupol** [comment]
Will do !

File: tests/phpunit/Environment/BuildContextResolverTest.php [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+use OndraM\CiDetector\CiDetector;
+use OndraM\CiDetector\Exception\CiNotDetectedException;
+use OndraM\CiDetector\TrinaryLogic;
+use PHPUnit\Framework\TestCase;
+
+final class BuildContextResolverTest extends TestCase
+{
+    public function test_resolve_throws_when_ci_could_not_be_detected(): void
+    {
+        $ciDetector = $this->prophesize(CiDetector::class);
```
Comment (theofidry) [inline tests/phpunit/Environment/BuildContextResolverTest.php]: should be doable with `Mock` right?

File: tests/phpunit/Environment/BuildContextResolverTest.php [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+        $this->expectExceptionMessage('The branch name could not be determined for the current process');
+
+        $buildContextResolver->resolve();
+    }
+
+    public function provideBlankOrEmptyString(): Generator
+    {
+        yield 'string-blank' => [' '];
+
+        yield 'string-empty' => [' '];
```
Comment (theofidry) [inline tests/phpunit/Environment/BuildContextResolverTest.php]: this look incorrect

File: tests/phpunit/Environment/BuildContextResolverTest.php (hunk 1 above)
Comment (localheinz) [reply tests/phpunit/Environment/BuildContextResolverTest.php]: Adjusted!

File: tests/phpunit/Environment/BuildContextResolverTest.php (hunk 2 above)
Comment (localheinz) [reply tests/phpunit/Environment/BuildContextResolverTest.php]: Thank you, good catch!

**Comment by maks-rafalko** [comment]
Thank you @localheinz

**Comment by localheinz** [comment]
Thank you, @maks-rafalko and @theofidry!

I will update `infection/site` as well!

**Comment by localheinz** [comment]
@maks-rafalko 

I have updated the pull request at https://github.com/infection/site/pull/155!

**Pull request merged**